feat(hint): added hasNoOffset prop (#10488)
patternfly/patternfly-react: packages/react-core/src/components/Hint/examples/Hint.md

```ts file="HintBasicWithoutTitle.tsx"

```

> ### Basic without no actions offset

Reviewer: ```suggestion
### With actions
```
Do you think this could just be called "with actions", since that's the ultimate concept that the example is supporting? Or is it important to specifically call out the prop here

Author: I think this makes sense 👍

You can add actions to a hint to allow users to respond to suggestions and initiate any related processes.

> When a `<Hint>` contains `actions`, a negative margin is applied to the actions for proper alignment between the hint title and the actions. For larger-sized titles or non-plain actions such as a button, remove this negative margin by setting `hasNoActionsOffset` to `true`.

Reviewer: ```suggestion
When a `<Hint>` contains `actions`, a negative margin is applied to the actions so that they are better aligned with the hint title. For larger-sized titles or non-plain actions (such as a button), remove this negative margin by setting `hasNoActionsOffset` to `true`.
``` 
nitpicky but maybe this so there's one less "actions"?

Author: Wow yes thanks, reading it back now all i see is the word "actions" so this change is worth it 👏

> The `actions` property for `<Hint>` can optionally be combined with the `hasNoOffset` property, which is `false` by default. When `hasNoOffset` is `false`, a negative margin is applied to help align default-sized hint titles with hint actions.
> 
> You may use `hasNoOffset` to remove this negative margin, which better aligns hint actions in implementations that use large hint titles, for example.
> 
> Select the "actions hasNoOffset" checkbox in the example below to illustrate this behavior.

Reviewer: ```suggestion
You can add actions to a hint to allow users to respond to suggestions and initiate any related processes.

When a `<Hint>` contains `actions`, use the `hasNoOffset` property to ensure that there is proper alignment between the hint title and the actions component (such as a button). For default-sized titles, pass in the `hasNoOffset` property (set to `false` by default) to apply a negative margin to the hint. For larger-sized titles, remove this negative margin by setting `hasNoOffset` to `true`.

To illustrate this behavior, select the "actions hasNoOffset" checkbox in the following example.
```
wdyt? too wordy?

Author: This is a confusing prop so I think the description helps, even if lengthy.  The only change I'd propose is this part:

> For default-sized titles, pass in the `hasNoOffset` property (set to `false` by default) to apply a negative margin to the hint.

The `hasNoOffset` (potentially renaming to `hasNoActionsOffset`) prop is false by default like you mention, which means that for default-sized titles the user doesn't need to think about it at all or do anything.  For this reason, I think we can call that out or simply remove this sentence and only point out how to use the prop if they actually want to change the default behavior.

Author: @edonehoo I trimmed this down a little with my suggestions above, let me know what you think - open to further changes:
- https://patternfly-react-pr-10488.surge.sh/components/hint#with-actions

Reviewer: yes, great suggestion! I suggested a small tweak, but I also like it how it is
